src/Algebra/Graph/Acyclic/Labelled/Algorithm.hs

```diff
@@ -0,0 +1,35 @@
+module Algebra.Graph.Acyclic.Labelled.Algorithm where
+
+import Algebra.Graph.Acyclic.Labelled.AdjacencyMap
+import qualified Algebra.Graph.AdjacencyMap.Algorithm as AM
+import Algebra.Graph.Label
+import qualified Algebra.Graph.Labelled.AdjacencyMap as LAM
+import Data.Map.Strict (Map, (!))
+import qualified Data.Map.Strict as Map
+import Data.Maybe (fromMaybe)
+
+-- TODO: Replace this function with 'skeleton' from Acyclic.Labelled to remove the use of fromMaybe
+-- TODO: Make 'topSort' more efficient
+-- TODO: Add examples and tests
+topSort :: (Ord a) => AdjacencyMap e a -> [a]
+topSort = fromMaybe [] . AM.topSort . LAM.skeleton . fromAcyclic
+
+-- TODO: Add time complexity
+-- TODO: Add examples using 'Optimum' data type
+-- | Compute the /shortest path/ to each vertex in the graph
+-- from a given source vertex.
+--
+-- The following examples assume that the edges are distances,
+-- ie. the edge 'Semiring' is 'Distance'.
+-- @
+-- dijkstra ('LAM.toAcyclicOrd' $ 'LAM.edges' [(2, 'b', 'c'), (1, 'a', 'b'), (3, 'a', 'c')]) 'z' == Map.'Map.fromList' [('z', 0), ('a', Infinite), ('b', Infinite), ('c', Infinite)]
+-- dijkstra ('LAM.toAcyclicOrd' $ 'LAM.edges' [(2, 'b', 'c'), (1, 'a', 'b'), (3, 'a', 'c')]) 'a' == Map.'Map.fromList' [('a', 0), ('b', 1), ('c', 3)]
+-- @
+optimumPath :: (Semiring e, Ord a) => AdjacencyMap e a -> a -> Map a e
+optimumPath am s = foldl (relaxVertex em) (initialize em) vL
```

Comment:
Are you sure you don't need a strict version of `foldl` here, as explained [here](https://wiki.haskell.org/Foldr_Foldl_Foldl')?

Reply:
Ah, yes, I think using a strict version of foldl would be better, considering that the library is made to work with huge graphs.